var/spack/repos/builtin/packages/pmix/package.py

```diff
@@ -154,40 +154,43 @@ class Pmix(AutotoolsPackage):
         deprecated=True,
     )
 
-    depends_on("c", type="build")  # generated
-
-    variant(
-        "pmi_backwards_compatibility",
-        default=True,
-        description="Toggle pmi backwards compatibility",
-    )
-
+    variant("docs", default=False, when="@master", description="Build documentation")
+    variant("python", default=False, when="@4.1.2:", description="Enable Python bindings")
     variant(
         "restful",
         default=False,
         when="@4:",
-        description="Allow a PMIx server to request services from " "a system-level REST server",
+        description="Allow a PMIx server to request services from a system-level REST server",
+    )
+    variant(
+        "pmi_backwards_compatibility",
+        default=True,
+        when="@1.2.5:3",
+        description="Enable PMI backwards compatibility",
     )
 
-    variant("python", default=False, when="@4.1.2:", description="Enable python bindings")
-
-    variant("docs", default=False, description="Build manpages")
-
+    depends_on("c", type="build")
+    depends_on("pkgconfig", type="build")
     depends_on("m4", type="build", when="@master")
-    depends_on("autoconf", type="build", when="@master")
-    depends_on("automake", type="build", when="@master")
-    depends_on("libtool", type="build", when="@master")
+    depends_on("autoconf@2.69:", type="build", when="@master")
+    depends_on("automake@1.13.4:", type="build", when="@master")
+    depends_on("libtool@2.4.2:", type="build", when="@master")
+    depends_on("flex@2.5.39:", type="build", when="@master")
     depends_on("perl", type="build", when="@master")
-    depends_on("pandoc", type="build", when="+docs")
-    depends_on("pkgconfig", type="build")
+    depends_on("python@3.7:", type="build", when="+docs")
+    depends_on("py-sphinx@5.0.0:", type="build", when="+docs")
+    depends_on("py-recommonmark", type="build", when="+docs")
+    depends_on("py-docutils", type="build", when="+docs")
+    depends_on("py-sphinx-rtd-theme", type="build", when="+docs")
 
     depends_on("libevent@2.0.20:")
-    depends_on("hwloc@1.0:1", when="@:2")
     depends_on("hwloc@1.11:1,2:", when="@3:")
+    depends_on("hwloc@1.0:1", when="@:2")
```

Comment:
Not sure I understand this rule - we require hwloc 1.11 as a minimum for all PMIx library versions that require hwloc at all (some do not).

Reply:
Okay, updated to simplify the syntax. For pmix versions 2 or lower, a hwloc version 1.x.x is required. For pmix versions 3 or higher, a hwloc version 1.11.x or higher is required.

Reply:
Hmmm...sorry, I don't think I made myself clear. Versions 2 and below don't have any dependency on hwloc. We didn't start involving hwloc until v3. It probably doesn't matter as nobody should be using something that old anyway - just clarifying for correctness.

Reply:
Looking at the commit history, that hwloc dependency for versions 2 and lower was added as a kind of safety feature. Previous discussion here: https://github.com/spack/spack/pull/30897